docs/meta/project-governance.md

```diff
@@ -0,0 +1,136 @@
+# Project Governance
+
+This document describes how the PyPDF2 project is managed. It describes the
+different actors, their roles, and the responsibilities they have.
+
+## Terminology
+
+* The **project** is PyPDF2 - a free and open-source pure-python PDF library
+capable of splitting, merging, cropping, and transforming the pages of PDF files.
+  It includes the [code, issues, and disscussions on GitHub](https://github.com/py-pdf/PyPDF2),
+  and [the documentation on ReadTheDocs](https://pypdf2.readthedocs.io/en/latest/),
+  [the package on PyPI](https://pypi.org/project/PyPDF2/), and
+  [the website on GitHub](https://py-pdf.github.io/PyPDF2/dev/bench/).
+* A **maintainer** is a person who has technical permissions to change one or
+  more part of the projects. It is a person who is driven to keep the project running
+  and improving.
+* A **contributor** is a person who contributes to the project. That could be
+  through writing code - in the best case through forking and creating a pull
+  request, but that is up to the maintainer. Other contributors describe issues,
+  help to ask questions on existing issues to make them easier to answer,
+  participate in discussions, and help to improve the documentation. Contributors
+  are similar to maintainers, but without technial permissions.
+* A **user** is a person who imports PyPDF2 into their code. All PyPDF2 users
+  are developers, but not developers who know the internals of PyPDF2. They only
+  use the public interface of PyPDF2. They will likely have less knowledge about
+  PDF than contributors.
+* The **community** is all of that - the users, the contributors, and the maintainers.
+
+
+## Governance, Leadership, and Steering PyPDF2 forward
+
+PyPDF2 is a free and open source project with over 100 contributors and likely
+(way) more than 1000 users.
+
+As PyPDF2 does not have any formal relationship with any company and no funding,
+all the work done by the community are voluntary contributions. People don't
+get paid, but choose to spend their free time to create software of which
+many more are profiting. This has to be honored and respected.
+
+Despite such a big community, the project was dormant from 2016 to 2022.
+There were still questions asked, issues reported, and pull requests created.
+But the maintainer didn't have the time to move PyPDF2 forward. During that
+time, nobody else stepped up to become the new maintainer.
+
+For this reason, PyPDF2 has the **Benevolent Dictator for Life** (BDFL)
+governance model. The BDFL is a maintainer with all technical permissions -
+most importantly the permission to push new PyPDF2 versions on PyPI.
```

Comment:
> The bus factor of 1 somewhat largely remains.

I don't see a way around this as long as there is not more community members who would be ready to step in. I completely agree that it would be good to have a bus factor larger than 1.

I see three scenarios how PyPDF2 development could get blocked again:

1. **BDFL looses interest**: Can happen and will likely happen. It would be good too have a clear process on what to do in that case. At the moment, I would probably ask you @MasterOdin if you want to become BDFL - but if there is nobody who is trustworthy + wants to spend the time on it, that's an issue.
2. **BDFL looses access** (PyPI and GitHub): I'm actually not sure how they handle this... but it seems rather unlikely to me.
3. **BDFL is dead**: Hopefully also unlikely 😅 However, Github has a [deceased user policy](https://docs.github.com/en/site-policy/other-site-policies/github-deceased-user-policy). I'm currently discussion with a couple of friends who would be that. I'm telling them that they should hand-over the project to the most active maintainer if that ever happens. For PyPI, I haven't seen anything like that.

Reply:
> The reasoning for using BDFL doesn't follow from the previous paragraph. 

My thoughts were: If we had many people who really cared about PyPDF2 (enough to become a maintainer), we could make something like a steering council. We have a lot of users, but not a lot of developers. And even less people who would like to become maintainers. That means to me that a more democratic approach cannot work.

Reply:
@MasterOdin I was just pointed to https://pypi.org/help/#project-name-claim - that means it is possible to transfer the PyPI ownership  if the project owner(s) on PyPI are not reachable for 1 year or longer.

It might be worth it to add a section in the docs how to deal with that / what to do in this case.

Reply:
I'd say I'm not complaining about the concept of a BFDL / BD, I'm totally on board with that, especially at the current stage of the project's lifecycle.

My only complaint was with this line specifically:
> The BDFL is a maintainer with all technical permissions - most importantly the permission to push new PyPDF2 versions on PyPI.

The previous setup was that mstamy2 was a singular maintainer / "BFDL", and that because he lost the time to take the project forward and was the only one with technical access to the project, it completely stalled. Saying that having a singular BDFL here doesn't exactly solve that problem where again, if you were to disappear without a trace, yes another maintainer could _maybe_ get appropriate access on GH, PyPI, etc. to start publishing, it's still a massive pain, and that all I'm asking is to make sure we have multiple active people in place with those project permissions, while again, having a BD who owns the technical side of things (veto power over any concept / issue, steering of the project, etc.).

I don't think this is something that needs to be solved right away, it's just food for thought.